File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.Arrays;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        
        // 몸무게 큰 사람과 작은 사람 짝지어서 빼기
        Arrays.sort(people);
        
        int left = 0;
        int right = people.length - 1;
        int boats = 0;
        
        while(left <= right) {
            if (people[left] + people[right] <= limit) {
                left++;
            }
            right--;
            answer++;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    answer = 0

    people.sort()
    
    left = 0
    right = len(people) - 1
    boats = 0
    
    while left <= right:
        if people[left] + people[right] <= limit:
            left += 1
            
        right -= 1
        boats += 1
    
    return boats